solr/core/src/java/org/apache/solr/core/SolrCore.java

```diff
@@ -2339,8 +2339,7 @@ public IndexFingerprint getIndexFingerprint(SolrIndexSearcher searcher, LeafRead
 
       final SolrIndexSearcher currSearcher = currSearcherHolder == null ? null : currSearcherHolder.get();
 
-      @SuppressWarnings({"rawtypes"})
-      Future future = null;
+      Future<Void> future = null;
```

Comment:
Yea, that's part of why I have the comment elsewhere that this could/should be an AtomicReference, but I don't want to change it to that pattern in this PR.